src/ext/integrations/integrations.c

```diff
@@ -1,8 +1,50 @@
 #include "integrations.h"
 
+#include "ddtrace_string.h"
 #include "elasticsearch.h"
 #include "test_integration.h"
 
+ddtrace_integration ddtrace_integrations[] = {
+    {DDTRACE_INTEGRATION_CAKEPHP, "CAKEPHP", ZEND_STRL("cakephp")},
+    {DDTRACE_INTEGRATION_CODEIGNITER, "CODEIGNITER", ZEND_STRL("codeigniter")},
+    {DDTRACE_INTEGRATION_CURL, "CURL", ZEND_STRL("curl")},
+    {DDTRACE_INTEGRATION_ELASTICSEARCH, "ELASTICSEARCH", ZEND_STRL("elasticsearch")},
+    {DDTRACE_INTEGRATION_ELOQUENT, "ELOQUENT", ZEND_STRL("eloquent")},
+    {DDTRACE_INTEGRATION_GUZZLE, "GUZZLE", ZEND_STRL("guzzle")},
+    {DDTRACE_INTEGRATION_LARAVEL, "LARAVEL", ZEND_STRL("laravel")},
+    {DDTRACE_INTEGRATION_LUMEN, "LUMEN", ZEND_STRL("lumen")},
+    {DDTRACE_INTEGRATION_MEMCACHED, "MEMCACHED", ZEND_STRL("memcached")},
+    {DDTRACE_INTEGRATION_MONGO, "MONGO", ZEND_STRL("mongo")},
+    {DDTRACE_INTEGRATION_MYSQLI, "MYSQLI", ZEND_STRL("mysqli")},
+    {DDTRACE_INTEGRATION_PDO, "PDO", ZEND_STRL("pdo")},
+    {DDTRACE_INTEGRATION_PREDIS, "PREDIS", ZEND_STRL("predis")},
+    {DDTRACE_INTEGRATION_SLIM, "SLIM", ZEND_STRL("slim")},
+    {DDTRACE_INTEGRATION_SYMFONY, "SYMFONY", ZEND_STRL("symfony")},
+    {DDTRACE_INTEGRATION_WEB, "WEB", ZEND_STRL("web")},
+    {DDTRACE_INTEGRATION_WORDPRESS, "WORDPRESS", ZEND_STRL("wordpress")},
+    {DDTRACE_INTEGRATION_YII, "YII", ZEND_STRL("yii")},
+    {DDTRACE_INTEGRATION_ZENDFRAMEWORK, "ZENDFRAMEWORK", ZEND_STRL("zendframework")},
+};
+size_t ddtrace_integrations_len = 0;
+
+// Map of lowercase strings to the ddtrace_integration equivalent
+static HashTable _dd_string_to_integration_name_map;
+
+static void _dd_add_integration_to_map(char* name, size_t name_len, ddtrace_integration* integration);
+
+void ddtrace_integrations_minit(void) {
+    ddtrace_integrations_len = sizeof ddtrace_integrations / sizeof ddtrace_integrations[0];
```

Comment:
This can be done up above on line 28.

Reply:
Thanks for pointing this out. My original thought was `ddtrace_integrations_len == 0` means that the integrations have not been set up yet so that you could check if the `_dd_string_to_integration_name_map` had been initialized. But since that is private, we probably don't need this. Plus I can see how giving `ddtrace_integrations_len == 0` semantics is confusing. So I moved it per your suggestion. :)